Sources/XCTest/Public/XCTestCase.swift

```diff
@@ -108,41 +130,128 @@ open class XCTestCase: XCTest {
     /// - Note: Like the designated initializer for Apple XCTest's XCTestCase,
     ///   `-[XCTestCase initWithInvocation:]`, it's rare for anyone outside of
     ///   XCTest itself to call this initializer.
-    public required init(name: String, testClosure: @escaping XCTestCaseClosure) {
+    @available(*, deprecated, message: "Use init(name:closureType:) instead.")
+    public init(name: String, testClosure: @escaping XCTestCaseClosure) {
         _name = "\(type(of: self)).\(name)"
-        self.testClosure = testClosure
+        self.testClosure = .regular(testClosure)
     }
 
+    public required init(name: String, closureType: ClosureType) {
+        _name = "\(type(of: self)).\(name)"
+        self.testClosure = closureType
+    }
     /// Invoking a test performs its setUp, invocation, and tearDown. In
     /// general this should not be called directly.
     open func invokeTest() {
-        performSetUpSequence()
```

Comment:
I think we ought to reintroduce the performSetUpSequence and performTearDownSequence helpers here, since they make the overall flow through this method a bit clearer. What do you think?